### PR TITLE
Ubuntu 22.04, second batch

### PR DESCRIPTION
**Pull request by nobodysu**

### CODE REVIEW

**Comment by roddhjav** [comment]
As I have now added support for dbus for most system services, could you test again your modification from the last changes?

**Comment by nobodysu** [comment]
Should I drop `peer=(name=...)`?

**Comment by roddhjav** [comment]
No, I use it myself sometime.

**Comment by nobodysu** [comment]
Could we use one of the following formatting? Because with current one important parameters got visually lost in the indentation.
```
  dbus (receive) bus=session
       path=/org/freedesktop/Notifications interface=org.freedesktop.Notifications member=NotificationClosed peer=(name=:*),
```
or
```
  dbus (receive) bus=session
       path=/org/freedesktop/Notifications
       interface=org.freedesktop.Notifications
       member=NotificationClosed
       peer=(name=:*),
```

File: apparmor.d/groups/freedesktop/polkitd [hunk 1]
```diff
@@ -22,14 +22,30 @@ profile polkitd @{exec_path} {
 
   ptrace (read),
 
-  dbus (send,receive) bus=system path=/org/freedesktop/PolicyKit[0-9]/*
-       interface=org.freedesktop.{DBus.Introspectable,DBus.Properties,PolicyKit[0-9].*},
-
-  dbus send bus=system path=/org/freedesktop/DBus
+  dbus (send) bus=system path=/org/freedesktop/DBus
```
Comment (nobodysu) [inline apparmor.d/groups/freedesktop/polkitd:25]: The idea here is to granularize and allow sending all members, but receive only mentioned.

**Comment by nobodysu** [comment]
What's the purpose of conditional `@{libexec}`? It brings additional complexity and requires a configure script while not covering all cases, e.g.:
```
*symlinks ommited*
xubuntu2204$ ls -l ...
/usr/libexec/polkit-agent-helper-1
/usr/libexec/gvfsd-admin
/usr/lib/update-notifier/package-system-locked
/lib/update-notifier/package-system-locked
```
Isn't `+=` more appropriate for `@{libexec}` tunable?

**Comment by roddhjav** [comment]
I did it to ensure not allowing run from `/usr/lib` when the lib in is `/usr/libexec`. Ultimately it would be possible to have something like `@{libexec}=/{usr/,}lib{,exec}/` but that would require to change all `/{usr/,}lib` by `@{libexec}` which raises even more changes and complexity.

> it requires a configure script

A configure/build script is still needed anyway, so that does not change that much

All lib in your example are actually correctly managed by the current profile.

File: apparmor.d/groups/freedesktop/polkitd (hunk 1 above)
Comment (roddhjav) [reply apparmor.d/groups/freedesktop/polkitd:25]: That's look like a pain to manage all of this for all profiles.

I mean, in concept `polkitd` 'own' the `/org/freedesktop/PolicyKit[0-9]` namespace in dBus. Therefore it makes sense to allow everything in it regardless send or receive. (cf the systemd polkit service definition that does the same).

It is also a simplification as a change in the program would not change the profile.

Comment (nobodysu) [reply apparmor.d/groups/freedesktop/polkitd:25]: Alright, that's reasonable. I hope send/receive parenthesis is OK.

Comment (roddhjav) [reply apparmor.d/groups/freedesktop/polkitd:28]: As you might have seen, I usually do not put them unless it is mandatory, but honestly we have other priorities than tracking parenthesis.

**Comment by roddhjav** [comment]
Thank, for the PR. Merged!

**Pull request merged**